arch/xtensa/core/mpu.c

```diff
@@ -0,0 +1,755 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/spinlock.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/arch/xtensa/arch_inlines.h>
+#include <zephyr/arch/xtensa/mpu.h>
+#include <zephyr/linker/linker-defs.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <xtensa/corebits.h>
+#include <xtensa/config/core-matmap.h>
+#include <xtensa/config/core-isa.h>
+#include <xtensa_mpu_priv.h>
+
+extern char _heap_end[];
+extern char _heap_start[];
+extern char __data_start[];
+extern char __data_end[];
+
+__weak const struct xtensa_mpu_range xtensa_soc_mpu_ranges[0];
+__weak int xtensa_soc_mpu_ranges_num;
```

Comment:
Not really my fight to pick, but for sure the Official Way To Manage Memory Maps in Zephyr is now in Devicetree, with a whole schema and everything.   This doesn't bug me personally but I guarantee someone (@carlocaione @stephanosio seem like good candidates) is going to hate it.

Reply:
For now, it is simpler to do it this way without cluttering the code with devicetree macros. Using devicetree memory regions is a future enhancement.

I think only ARM MPU is using devicetree to describe memory regions. ARM MMU does not seem to do it. ARC MPU does not do it either. I think all MMU code are also not using devicetree region.